OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -37,24 +37,27 @@ MarkersReference::MarkersReference() :
 }
 
 MarkersReference::MarkersReference(const std::string& markerFile,
+                                   const Set<MarkerWeight>* markerWeightSet,
                                    Units modelUnits) :
     MarkersReference() {
-    loadMarkersFile(markerFile, modelUnits);
+    loadMarkersFile(markerFile, markerWeightSet, modelUnits);
 }
 
 MarkersReference::
 MarkersReference(const TimeSeriesTable_<SimTK::Vec3>& markerTable,
                  const Set<MarkerWeight>* markerWeightSet,
                  Units units) :
     MarkersReference() {
+    // Make a writable copy of the marker table owned by this reference.
     _markerTable = markerTable;
     if(markerWeightSet != nullptr)
         upd_marker_weights() = *markerWeightSet;
-    populateFromMarkerData(markerTable, units.getAbbreviation());
+    populateFromMarkerData(_markerTable, markerWeightSet, units.getAbbreviation());
 }
 
 void MarkersReference::loadMarkersFile(const std::string markerFile,
-                                       Units modelUnits) {
+                            const Set<MarkerWeight>* markerWeightSet,
+                            Units modelUnits) {
     auto fileExt = FileAdapter::findExtension(markerFile);
```

Comment:
From a user perspective, it would be confusing for a method called loadMarkersFile  (that is supposed to take a file and create a table) to also take `Set<MarkerWeight>*`, can we refactor/rename it so the loading is just simple reading then operations are performed on result? Maybe also rename method populateFromMarkerData to something like populateFromDataAndWeights since Markers is implicit in the class name.

Reply:
If a user wants to read a Table of Marker data they should NOT use a `MarkersReference` for that. We could rename `loadMarkersFile` to `initializeFromMarkerFile` but the method serves the `MarkersReference`-- it isn't a `DataAdapter`. I can refactor to make this method private and force Tools to use the constructor. Is that your preference?

Reply:
Your call @aseth1 I think renaming would make the most sense because it's in the public interface.

Reply:
OK I renamed to `initializeFromMarkersFile` and left it in the public interface although it is only used by `InverseKinematicsTool`.

Reply:
👍